lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -3093,6 +3093,58 @@ LogicalResult ConvertAtenOp<AtenBroadcastToOp>::matchAndRewrite(
       "unimplemented: broadcasts other than same rank or zero ranked tensor.");
 }
 
+template <>
+LogicalResult ConvertAtenOp<AtenQuantizePerTensorOp>::matchAndRewrite(
+    AtenQuantizePerTensorOp op, OpAdaptor adaptor,
+    ConversionPatternRewriter &rewriter) const {
+
+  auto resultTy = getTypeConverter()
+                      ->convertType(op->getOperand(0).getType());
+
+  // This is dead and will be removed
```

Comment:
I'm guessing that this has to do with the pattern supported in `AtenIntReprOp`? I think cookie-trail comments are needed to help people understand and maintain the approach.

Reply:
Yes, I can explain this better. It feels a bit hacky, but essentially:
In `ConvertAtenOp<AtenIntReprOp>`, we replace
```
a = "torch.constant.int"() {value = 0 : i64} : () -> !torch.int
y = AtenQuantizePerTensorOp(x, offset=a)
z = AtenIntReprOp(y)
```
by 
```
a = "torch.constant.int"() {value = 0 : i64} : () -> !torch.int
y = AtenQuantizePerTensorOp(x, offset=a)
z = tosa.cast(x) // note the 'x' here
```
Now a, and y are dead. They are not removed during the TorchToTosa pass, and then it fails with
`error: failed to legalize operation 'torch.constant.int'`.

To work around this, I replace `AtenQuantizePerTensorOp` by tosa.cast, which magically gets rid of `a`,
and later this dead cast is removed by some other pass.

Reply:
Ok, I rewrote that part to instead:
1) replace AtenQuantizePerTensorOp by cast to int8
2) replace AtenIntReprOp by its operand (i.e. make it no-op)
This is much cleaner now.